pkg/build/controller/buildpod/factory.go

```diff
@@ -0,0 +1,213 @@
+package buildpod
```

Comment:
not obvious to me why we have a "factory" here, seems like all this content should just be moved into controller.go.